addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -292,7 +292,11 @@ func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionRes
 	}
 	apiResourceList, err := wh.cachedDiscoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, err
+		// Sometimes it is unable to retrieve the complete list of server APIs i.g. an aggregated API server is down,
+		// ErrGroupDiscoveryFailed is returned in this case. We can still get other API resources and ignore this error.
+		if _, ok := err.(*discovery.ErrGroupDiscoveryFailed); !ok {
```

Comment:
This check is in the wrong function. 
This function (getGVR) one and only purpose is to return a groupversionresource, or to return error. 

"func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionResource, error)"

Make the check after the call to this. here: https://github.com/vmware-tanzu/tanzu-framework/blob/ba6b6f70b1873e1fbc0fb00f9b8eb47eb42367ee/addons/webhooks/clusterbootstrap_webhook.go#L260

for example:
```
		if err != nil &&  !err.(*discovery.ErrGroupDiscoveryFailed) {
			return field.Invalid(fldPath.Child("ProviderRef"), valuesFrom.ProviderRef, err.Error())
		}
```
I think that would make it more ovious and perhaps not even require  log message, but it would be nice just a couple more lines.

Reply:
If not, then you need too make sure apiResourceList is not nil